realm/realm-library/src/main/java/io/realm/RealmQuery.java

```diff
@@ -1092,6 +1093,28 @@ public boolean isValid() {
         return this;
     }
 
+    /**
+     * Returns a distinct set of objects of a specific class. As a Realm is unordered, it is undefined which objects are
+     * returned in case of multiple occurrences.
+     *
+     * @param fieldName the field name.
+     * @return a non-null {@link RealmResults} containing the distinct objects.
+     * @throws IllegalArgumentException if a field name does not exist or the field is not indexed.
+     */
+    public RealmResults<E> distinct(String fieldName) {
+        Table table = this.table.getTable();
+        long columnIndex = table.getColumnIndex(fieldName);
+        if (columnIndex == -1) {
+            throw new IllegalArgumentException(String.format("Field name '%s' does not exist.", fieldName));
+        }
+        TableView tableView = this.query.findAll().getDistinctView(columnIndex);
+        RealmResults<E> realmResults = RealmResults.createFromTableOrView(this.realm, tableView, this.clazz);
+        if (realm.handlerController != null) {
```

Comment:
I think this should be moved to the RealmResults constructor. It seems way to easy to forget to add it.